### ISSUE TITLE
Using External Tool on existing connection causes creation of 'New Connection' entry.

### ISSUE THREAD

**Issue by damiankaczkowski** [mRemoteNG/mRemoteNG#530]
Hello.

I would like to report 100% reproducible and annoying bug.  When one use external tool on existing connection then mRemoteNG creates "New Connection" entry with the name of used external tool entry name.

Steps to reproduce:
1. Create new SSH 2 connection (Crtl +N) and save it for eg. as 'test_connection'.
2.  Create new external tool entry. Eg:
  Display Name: 'test_external_tool'
  Filename: '.\putty\putty.exe'
  Arguments: '-load "Default Settings" %Hostname%'
  Try to integrate: yes
3. Connect to ssh server using newly created 'test_connection' entry.
4. Execute 'test_external_tool' on connected 'test_connection' window (test_connection must be already connected when executing external tool).

Check 'Connections' manager. You should see new connection entry named 'test_external_tool' being created on every external tool launch.

Greets.

|||
|--:|---|
|Operating system | Windows 10 x64 |
|mRemoteNG version| 1.75 hotfix 5  |

**Comment by sparerd** [comment]
I'm able to reproduce the issue. I'll try to get a fix in place this week

**Comment by sparerd** [comment]
Found roughly where the issue is. Stack trace showing the erroneous tree node addition on step 4 from the OP:

```
mRemoteNG.exe!mRemoteNG.Container.ContainerInfo.AddChildAt(mRemoteNG.Connection.ConnectionInfo newChildItem, int index) Line 62
mRemoteNG.exe!mRemoteNG.Container.ContainerInfo.AddChild(mRemoteNG.Connection.ConnectionInfo newChildItem) Line 41
mRemoteNG.exe!mRemoteNG.Connection.ConnectionInfo.SetParent(mRemoteNG.Container.ContainerInfo parent) Line 132
mRemoteNG.exe!mRemoteNG.Connection.ConnectionInfo.Clone() Line 81
mRemoteNG.exe!mRemoteNG.Tools.ExternalTool.GetAppropriateInstanceOfConnectionInfo() Line 101
mRemoteNG.exe!mRemoteNG.Tools.ExternalTool.BuildConnectionInfoForIntegratedApp() Line 94
mRemoteNG.exe!mRemoteNG.Tools.ExternalTool.StartIntegrated() Line 83
mRemoteNG.exe!mRemoteNG.Tools.ExternalTool.Start(mRemoteNG.Connection.ConnectionInfo startConnectionInfo) Line 49
mRemoteNG.exe!mRemoteNG.UI.Controls.ConnectionContextMenu.StartExternalApp(mRemoteNG.Tools.ExternalTool externalTool) Line 787
mRemoteNG.exe!mRemoteNG.UI.Controls.ConnectionContextMenu.OnExternalToolClicked(object sender, System.EventArgs e) Line 779
```

**Comment by sparerd** [comment]
resolved in commit f0faec9

we will include this in the next hotfix update

**Comment by schittli** [comment]
This message was created automatically by mail delivery software.

A message that you sent could not be delivered to one or more of its
recipients. This is a temporary error. The following address(es) deferred:

  thomas.schittli@akros.ch
    Domain jig.ch has exceeded the max emails per hour (246/200 (123%)) allowed.  Message will be reattempted later

------- This is a copy of the message, including all the headers. ------
Received: from o8.sgmail.github.com ([167.89.101.199]:56608)
	by rs111.nsresponse.com with esmtps (TLSv1.2:ECDHE-RSA-AES128-GCM-SHA256:128)
	(Exim 4.88)
	(envelope-from <bounces+848413-aac0-github=jig.ch@sgmail.github.com>)
	id 1d5a1k-000t7g-SN
	for github@jig.ch; Tue, 02 May 2017 10:47:59 -0500
DKIM-Signature: v=1; a=rsa-sha1; c=relaxed/relaxed; d=github.com; 
	h=from:reply-to:to:cc:in-reply-to:references:subject:mime-version:content-type:content-transfer-encoding:list-id:list-archive:list-post:list-unsubscribe; 
	s=s20150108; bh=CUl7BK1KGvlbRpiGDaNHm7OsWX4=; b=uzZxc6X3ZkooBE57
	dusYcAwN7M6uohv0qiYsW9Nrm6CimsL8+b3PH2XVPxzh7zlHb8SXxuVIf6YX/klb
	mimK8lHbK5euhCKDf/9VWombrFfnno9attYPKtp/PUJQOETGuCwFZSkQf5/rtl7a
	x4qjzQGbwOSu/lClFMMR/DtfWN8=
Received: by filter0281p1las1.sendgrid.net with SMTP id filter0281p1las1-10561-5908A9FB-5A
        2017-05-02 15:47:07.543466155 +0000 UTC
Received: from github-smtp2a-ext-cp1-prd.iad.github.net (github-smtp2a-ext-cp1-prd.iad.github.net [192.30.253.16])
	by ismtpd0002p1iad1.sendgrid.net (SG) with ESMTP id ynUO64C3R0GgIUZ_DLLdwQ
	for <github@jig.ch>; Tue, 02 May 2017 15:47:07.340 +0000 (UTC)
Date: Tue, 02 May 2017 08:47:07 -0700

**Comment by schittli** [comment]
This message was created automatically by mail delivery software.

A message that you sent could not be delivered to one or more of its
recipients. This is a temporary error. The following address(es) deferred:

  thomas.schittli@akros.ch
    Domain jig.ch has exceeded the max emails per hour (245/200 (122%)) allowed.  Message will be reattempted later

------- This is a copy of the message, including all the headers. ------
Received: from github-smtp2-ext4.iad.github.net ([192.30.252.195]:37095 helo=github-smtp2a-ext-cp1-prd.iad.github.net)
	by rs111.nsresponse.com with esmtps (TLSv1.2:ECDHE-RSA-AES256-GCM-SHA384:256)
	(Exim 4.88)
	(envelope-from <noreply@github.com>)
	id 1d5a1k-000t7h-Mr
	for github@jig.ch; Tue, 02 May 2017 10:47:54 -0500
Date: Tue, 02 May 2017 08:47:08 -0700
DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed; d=github.com;
	s=pf2014; t=1493740028;
	bh=xJEPTRgFzAgYl9+b76p+3zYu1z3daGQnA0XhZx98L70=;
	h=From:Reply-To:To:Cc:In-Reply-To:References:Subject:List-ID:
	 List-Archive:List-Post:List-Unsubscribe:From;
	b=ELSDUD6H4AjCKYKYp+ycVcj9gHyvqYkI14EGSqD1y2jvJML+wnPBczufU6gF/s8UM
	 pldiZIpMhw1NYkcJ/k+6csda0LY8tdRqxFp9jogahkwro7iBJHPHyiQypbp5igTeA/
	 bspmyKZLLqwfUizpfRGvYczincP9FijAdkWFvk30=

**Comment by kmscode** [comment]
@sparerd Could you please confirm... Looks like this was fixed in
commit: 692b26622cc69f351455fecc9871e0b1e2ecee9d

Not: f0faec9defdd5c79d6e53fd627ee81afba586197

I'd like to get this fix into the [hotfix6 branch](https://github.com/mRemoteNG/mRemoteNG/tree/hotfix6) so we can do a release soon.

**Comment by sparerd** [comment]
Yep i think you are right. Must have mis-copied it

**Comment by kmscode** [comment]
Confirmed 692b26622cc69f351455fecc9871e0b1e2ecee9d was committed to master. This will be included in 1.75 HF6
